java-checks/src/main/java/org/sonar/java/checks/spring/TransactionalMethodVisibilityCheck.java

```diff
@@ -38,14 +49,20 @@ public List<Tree.Kind> nodesToVisit() {
   @Override
   public void visitNode(Tree tree) {
     MethodTree method = (MethodTree) tree;
-    if (!method.symbol().isPublic() && hasTransactionalAnnotation(method)) {
-      reportIssue(method.simpleName(), "Make this method \"public\" or remove the \"@Transactional\" annotation");
+    if (!method.symbol().isPublic()) {
+      for(String annSymbol : proxyAnnotations){
+        if(hasAnnotation(method, annSymbol)){
+          reportIssue(
+            method.simpleName(),
+            "Make this method \"public\" or remove the \""+ annShortName.get(annSymbol)+"\" annotation");
```

Comment:
Let's add a period  at the end of the message (see [Official guideleines](https://docs.sonarsource.com/sonarqube/latest/extension-guide/adding-coding-rules/#messages)
```suggestion
            "Make this method \"public\" or remove the \""+ annShortName.get(annSymbol)+"\" annotation.");
```